tests/spin/test_lattice.py

```diff
@@ -539,3 +541,72 @@ def test_shape_error():
     lattice = "Octagon"
     with pytest.raises(ValueError, match="Lattice shape, 'Octagon' is not supported."):
         _generate_lattice(lattice=lattice, n_cells=n_cells)
+
+
+def test_neighbour_order_error():
+    r"""Test that an error is raised if neighbour order is greater than 1 when custom_edges are provided."""
```

Comment:
So we cannot have `custom_edges` for next-nearest neighbours with the current `Lattice` implementation?